### PR TITLE
Simplify the screen templates

### PR DESCRIPTION
**Pull request by stefanceriu**

### CODE REVIEW

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenCoordinator.swift [hunk 1]
```diff
@@ -17,15 +17,10 @@
...
-struct TemplateScreenCoordinatorParameters {
-    let promptType: TemplateScreenPromptType
-}
+struct TemplateScreenCoordinatorParameters { }
 
 enum TemplateScreenCoordinatorAction {
-    case accept
-    case cancel
-    
-    // Consider adding CustomStringConvertible conformance if the actions contain PII
```
Comment (pixlwave) [inline Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenCoordinator.swift:28]: Lets keep this comment in please.

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenCoordinator.swift [hunk 2]
```diff
@@ -41,18 +36,18 @@ final class TemplateScreenCoordinator: CoordinatorProtocol {
...
         self.parameters = parameters
         
-        viewModel = TemplateScreenViewModel(promptType: parameters.promptType)
+        viewModel = TemplateScreenViewModel()
     }
     
     func start() {
         viewModel.actions.sink { [weak self] action in
+            // swiftlint:disable:next print_deprecation
+            print("Coordinator: received view model action: \(action)")
```
Comment (pixlwave) [inline Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenCoordinator.swift:45]: Why print and not `MXLog` 🤨

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenModels.swift [hunk 3]
```diff
@@ -16,50 +16,32 @@
...
+}
+
+struct TemplateScreenViewStateBindings {
+    var composerText: String {
+        didSet {
+            if composerText != oldValue {
+                // swiftlint:disable:next print_deprecation
+                print("Bidings: composer text changed to: \(composerText)")
+            }
+        }
```
Comment (pixlwave) [inline Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenModels.swift:38]: Is the didSet useful here? Also same comment on print. Also also, "Bidings"

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift [hunk 4]
```diff
@@ -17,88 +17,36 @@
...
-            
-            Image(systemName: context.viewState.promptType.imageSystemName)
-                .resizable()
-                .scaledToFit()
-                .frame(width: 100)
-            
-            HStack {
-                Text("Counter: \(context.viewState.count)")
-                    .font(.compound.bodyLG)
+        NavigationStack {
```
Comment (pixlwave) [inline Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift:23]: We probably shouldn't include the stack in the template, just in the preview :)

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenCoordinator.swift (hunk 1 above)
Comment (stefanceriu) [reply Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenCoordinator.swift:28]: Ups, sorry, didn't mean to remove it

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift [hunk 5]
```diff
@@ -17,88 +17,36 @@
...
+            Form {
```
Comment (pixlwave) [inline Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift:24]: For good compound usage, we *must* wrap the `Form`s content in a `FormSection` with the `compoundFormSection()` modifier.

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift [hunk 6]
```diff
@@ -17,88 +17,36 @@
...
-                    .foregroundColor(counterColor)
-                
-                Button("−") {
-                    context.send(viewAction: .decrementCount)
-                }
-                .buttonStyle(.elementGhost())
-                
-                Button("+") {
-                    context.send(viewAction: .incrementCount)
+                    .foregroundColor(.compound.textActionAccent)
```
Comment (pixlwave) [inline Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift:27]: You shouldn't need this, the right colours are provided by the style.

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift [hunk 7]
```diff
@@ -17,88 +17,36 @@
...
-                .resizable()
-                .scaledToFit()
-                .frame(width: 100)
-            
-            HStack {
-                Text("Counter: \(context.viewState.count)")
-                    .font(.compound.bodyLG)
+        NavigationStack {
+            Form {
+                TextField(context.viewState.placeholder, text: $context.composerText)
```
Comment (pixlwave) [inline Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift:25]: Sadly the one downside of the textField is it also needs the `placeholder:` parameter with `Text(thePlaceHolder).compoundFormTextFieldPlaceholder()` for correct styling.

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift [hunk 8]
```diff
@@ -17,88 +17,36 @@
...
-                Button("−") {
-                    context.send(viewAction: .decrementCount)
-                }
-                .buttonStyle(.elementGhost())
-                
-                Button("+") {
-                    context.send(viewAction: .incrementCount)
+                    .foregroundColor(.compound.textActionAccent)
+                    .textFieldStyle(.compoundForm)
+                Button {
```
Comment (pixlwave) [inline Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift:29]:
```suggestion
                
                Button {
```

**Review by pixlwave — APPROVED** [review]
LGTM (despite the number of comments I added).

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenModels.swift (hunk 3 above)
Comment (stefanceriu) [reply Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenModels.swift:38]: I figured it's another way of accessing changes but I think you're right, better we don't showcase this

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenCoordinator.swift (hunk 2 above)
Comment (stefanceriu) [reply Tools/Scripts/Templates/SimpleScreenExample/ElementX/TemplateScreenCoordinator.swift:45]: Becuse MXLog isn't setup for UI tests but you're right, the templates should be clean

File: Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift (hunk 4 above)
Comment (stefanceriu) [reply Tools/Scripts/Templates/SimpleScreenExample/ElementX/View/TemplateScreen.swift:23]: Good point 👍

**Pull request merged**